### PR TITLE
CAKE-5184 | Handle posts and search for posts

### PR DESCRIPTION
**Pull request by vforge**
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5184
## Description

- Search for all posts, not only quizzes
- Have a fallback title for old posts that are without one
- Add poll icon

## Reviewers

@Wikia/cake

### CODE REVIEW

**Review by cschucode — APPROVED** [review]
LGTM

File: app/components/post-search-results-item.js [hunk 1]
```diff
@@ -6,6 +6,20 @@ import Thumbnailer from '../modules/thumbnailer';
...
 export default Component.extend({
   classNames: ['post-search-results-item'],
+
+  // some old posts do not have title, we'll fake it here
+  getTitle: computed('post', function () {
+    if (this.post.title) {
+      return this.post.title;
+    }
+
+    if (this.post.content.length > 100) {
```
Comment (JoshuaRogan) [inline app/components/post-search-results-item.js]: Any chance this can be null?

File: app/components/post-search-results.js [hunk 2]
```diff
@@ -168,18 +167,12 @@ export default Component.extend({
...
   update(state) {
     if (!this.isDestroyed) {
-      const results = state.results.map(item => (Object.assign({}, item, {
-        image: item.image,
-        stats: item.stats || {},
-        title: item.title,
-        type: item.type,
-        url: item.url,
-      })));
+      const results = state.results.map(item => extend({}, item));
```
Comment (JoshuaRogan) [inline app/components/post-search-results.js:170]: Good call 👍

**Review by JoshuaRogan — APPROVED** [review]
LGTM Maybe add a small null check but I think we are good.

File: app/components/post-search-results-item.js (hunk 1 above)
Comment (vforge) [reply app/components/post-search-results-item.js]: hmmm...

Comment (cschucode) [reply app/components/post-search-results-item.js]: Are you saying no post title and no post content?

**Pull request merged**